cmd/ocsp-updater/main.go

```diff
@@ -381,6 +395,64 @@ func (updater *OCSPUpdater) revokedCertificatesTick(batchSize int) error {
 	return nil
 }
 
+func internalsctToCT(reciept core.SignedCertificateTimestamp) (ct.SignedCertificateTimestamp, error) {
+	sig, err := ct.UnmarshalDigitallySigned(bytes.NewReader(reciept.Signature))
+	if err != nil {
+		return ct.SignedCertificateTimestamp{}, err
+	}
+	out := ct.SignedCertificateTimestamp{
+		SCTVersion: ct.Version(reciept.SCTVersion),
+		Timestamp:  reciept.Timestamp,
+		Extensions: reciept.Extensions,
+		Signature:  *sig,
+	}
+
+	err = out.LogID.FromBase64String(reciept.LogID)
+	if err != nil {
+		return ct.SignedCertificateTimestamp{}, err
+	}
+
+	return out, nil
+}
+
+func (updater *OCSPUpdater) generateOCSPSCTExtension(serial string) (pkix.Extension, error) {
+	output := make([]byte, 2)
+	for _, logID := range updater.logsIncludedInSCT {
+		receipt, err := updater.sac.GetSCTReceipt(serial, logID)
+		if err != nil {
+			continue
+		}
+
+		ctsct, err := internalsctToCT(receipt)
+		if err != nil {
+			continue
+		}
+
+		SCT, err := ct.SerializeSCT(ctsct)
+		if err != nil || len(SCT) == 0 || len(SCT) > 0xFFFF {
+			continue
+		}
+
+		binary.BigEndian.PutUint16(output, uint16(len(SCT)))
+		output = append(output, output[0], output[1])
+		output = append(output, SCT...)
+	}
+
+	if len(output) == 2 {
```

Comment:
The length here, and above where you initialize `output`, should be configurable.